pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -188,6 +256,66 @@ def _primitive_bind_call(cls, mps, wires, id=None):
             return type.__call__(cls, mps=mps, wires=wires, id=id)  # pragma: no cover
         return cls._primitive.bind(*mps, wires=wires, id=id)
 
+    def decomposition(self):  # pylint: disable=arguments-differ
+        filtered_hyperparameters = {
+            key: value for key, value in self.hyperparameters.items() if key != "input_wires"
+        }
+        return self.compute_decomposition(
+            self.parameters, wires=self.hyperparameters["input_wires"], **filtered_hyperparameters
+        )
+
+    @staticmethod
+    def compute_decomposition(mps, wires, work_wires):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+
+        Args:
+            mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
+                product of site matrices. See the usage details section for more information.
+
+            wires (Sequence[int]): wires that the template acts on
+            work_wires (Sequence[int]): list of extra qubits needed. The bond dimension of the mps
+                is defined as ``2^len(work_wires)``
+
+        Returns:
+            list[.Operator]: Decomposition of the operator
+        """
+
+        if work_wires is None:
+            raise AssertionError("To decompose MPSPrep you must specify `work_wires`.")
+
+        ops = []
+        n_wires = len(work_wires) + 1
+
+        for i, Ai in enumerate(mps):
+            vectors = []
+
+            # encodes the tensor Ai in a unitary matrix following Eq.23 in https://arxiv.org/pdf/2310.18410
+            if i == 0:
+                Ai = Ai.reshape((1, *Ai.shape))
+            elif i == len(mps) - 1:
+                Ai = Ai.reshape((*Ai.shape, 1))
+
+            for column in Ai:
+
+                vector = qml.math.zeros(2**n_wires, like=mps[0])
+
+                if qml.math.get_interface(mps[0]) == "jax":
+                    vector = vector.at[: len(column[0])].set(column[0])
+                    vector = vector.at[
+                        2 ** (n_wires - 1) : 2 ** (n_wires - 1) + len(column[1])
+                    ].set(column[1])
+
+                else:
+                    vector[: len(column[0])] = column[0]
+                    vector[2 ** (n_wires - 1) : 2 ** (n_wires - 1) + len(column[1])] = column[1]
+
+                vectors.append(vector)
+
+            matrix = _complete_unitary(vectors)
```

Comment:
If we could do this directly via QR decomposition, we might not need to separate it out.